go/services/reports/internal/services/reports_service.go

```diff
@@ -125,6 +131,114 @@ func (s *ReportsService) ScheduleReport(
 	return report, nil
 }
 
+func (s *ReportsService) PollReports(ctx context.Context) error {
+
+	pendingReports := make(map[string]bool)
+
+	for {
+		reports, err := s.reportRepository.GetPendingReports(ctx)
+
+		if err != nil {
+			s.logger.Error("Failed to get pending reports", zap.Error(err))
+			continue
+		}
+
+		for _, report := range reports {
+			if isAlreadyPending := pendingReports[report.Id]; isAlreadyPending == true {
+				continue
+			}
+
+			pendingReports[report.Id] = true
```

Comment:
Can't find where we remove reportId from the map after it's generated properly, do we do that?

Reply:
This was meant to just not await already "awaited" pending report, so in this case it doesn't matter if a report is completed. But, it's a good point to remove it from a map once it's done, so it doesn't grow infinitely.